acinclude.m4

```diff
@@ -631,14 +631,26 @@ if test "$GIFLIB_LIBS" = ""; then
   INCLUDES="$GIFLIB_INCLUDES $INCLUDES"
 
   # Try the standard search path first
-  AC_TRY_LINK([#include <gif_lib.h>],[EGifSetGifVersion(0,0)], [
+  AC_TRY_LINK([#include <gif_lib.h>],[
+    #if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR > 4
+    EGifSetGifVersion(0,0);
+    #else
+    GifLastError();
+    #endif
+  ], [
     GIFLIB_LIBS="-lgif"
```

Comment:
How about calling `DGifSlurp(0)` instead? This should compile on giflib4 and giflib5, and we only link, we don't execute, so it doesn't matter that it won't run.